src/bio.c

```diff
@@ -580,6 +605,7 @@ static int wolfSSL_BIO_MEMORY_write(WOLFSSL_BIO* bio, const void* data,
     bio->num = (int)bio->mem_buf->max;
     bio->wrSz += len;
     bio->wrIdx += len;
+    bio->bytes_written += (word32)len;
```

Comment:
I think its easier to implement `bytes_written` and `bytes_read` inside of `wolfSSL_BIO_read`, `wolfSSL_BIO_write`, `wolfSSL_BIO_nread`, and `wolfSSL_BIO_nwrite`. For example `wolfSSL_BIO_MD_write` and `wolfSSL_BIO_MD_read` are missing tracking.

Reply:
Yup indeed.  Done.  Was slightly tricky, btw, because `wolfSSL_BIO_read()` calls `wolfSSL_BIO_nread()` indirectly for the `WOLFSSL_BIO_BIO` case, and similarly for write.